pkg/gencosmosdb/cosmosdb/template.go

```diff
@@ -210,3 +216,47 @@ func (i *templateQueryIterator) Next(ctx context.Context) (templates *pkg.Templa
 
 	return
 }
+
+func (i *templateListIterator) Run(ctx context.Context, headers http.Header) (templates *interface{}, err error) {
```

Comment:
this shouldn't return `*interface{}` - it should take an `interface{}` as an argument and modify it

Reply:
I think the duplication can be eliminated here:
* change the existing Next() functions to be NextRaw() and, taking an interface{} argument instead of returning a *pkg.Templates
* add new Next() functions which instantiate a pkg.Templates, pass that into NextRaw() to be populated and return it as it does today without changing the signature of Next().

Reply:
see how do() does this as an example